olive/passes/onnx/conversion.py

```diff
@@ -553,3 +562,49 @@ def _run_for_config(
             converted_model_proto, str(Path(model.model_path).resolve().parent)
         )
         return model_proto_to_olive_model(converted_model_proto, output_model_path, config)
+
+
+def _convert_dynamic_shapes_to_torch_export_dims(
+    dynamic_shapes: Dict[str, Dict[int, torch.export.Dim]]
+) -> Dict[str, Dict[int, torch.export.Dim]]:
+    """Convert dynamic_shapes to torch export dims.
+
+    :param dynamic_shapes: the dynamic_shapes to convert
+    :return: the converted dynamic_shapes
+    """
+    if dynamic_shapes is None:
+        return None
+
+    # If the axes has the same name, they should be the same torch.export.Dim
+    torch_export_dim_farm: Dict[str, torch.export.Dim] = {}
+
+    # dynamic_shapes follows input format, which could be nested
+    def _from_tuple_to_dim(data: Union[Dict, List, Tuple, Any]) -> Union[Dict, List, Tuple, Any]:
+        if isinstance(data, dict):
+            for key, value in data.items():
+                data[key] = _from_tuple_to_dim(value)
+        elif isinstance(data, list):
+            for i in range(len(data)):
+                data[i] = _from_tuple_to_dim(data[i])
+        elif isinstance(data, tuple):
```

Comment:
I think you might have to treat tuple/list as the same here. In yaml and json, there is no tuple data type so it we have to use save and provide tuples using list.

Reply:
does the exporter care whether to original type wrapping the dynamic shapes are tuple or list? That might be a problem if so since we cannot differentiate between them in json/yaml.

Reply:
The idea is that the nested `dynamic_shapes` should follow the nested `input`, for example:

```python
        input = (
            torch.ones(5),
            (torch.zeros(5), torch.ones(5)),
            {"a": torch.zeros(5), "b": torch.ones(5)},
            torch.ones(4),
        )
        dim = torch.export.Dim("dim", min=3)
        dynamic_shapes = (
            {0: dim},
            ({0: dim}, {0: dim}),  # torch detects mismatches if we change this to list
            {"a": {0: dim}, "b": {0: dim}},
            None,
        )
```

When `model input` has tuple or list, the `dynamic_shapes` should follow.